jacobian_follower/src/jacobian_follower/jacobian_follower.cpp

```diff
@@ -812,3 +813,46 @@ PointSequence JacobianFollower::get_tool_positions(std::vector<std::string> tool
   std::transform(tool_transforms.cbegin(), tool_transforms.cend(), std::back_inserter(positions), get_translation);
   return positions;
 }
+Eigen::Matrix4Xd JacobianFollower::getLinkToRobotTransform(std::vector<std::string> joint_names,
+                                                           std::vector<double> joint_positions, std::string link_name) {
+  robot_state::RobotState state(model_);
+  state.setVariablePositions(joint_names, joint_positions);
+  auto const &transform = state.getGlobalLinkTransform(link_name);
+  return transform.matrix();
+}
+
+std::vector<Eigen::Matrix4Xd> JacobianFollower::getLinkToRobotTransforms(
+    std::vector<std::string> const &joint_names, std::vector<double> const &joint_positions,
+    robot_state::RobotStatePtr state, std::vector<std::string> const &link_names) const {
+  state->setVariablePositions(joint_names, joint_positions);
+  std::vector<Eigen::Matrix4Xd> transforms;
+  auto get_transform = [&](std::string const &link_name) {
+    auto const &transform = state->getGlobalLinkTransform(link_name);
+    return transform.matrix();
+  };
+  std::transform(link_names.cbegin(), link_names.cend(), std::back_inserter(transforms), get_transform);
+  return transforms;
+}
+
+std::vector<Eigen::Matrix4Xd> JacobianFollower::getLinkToRobotTransforms(
+    std::vector<std::string> const &joint_names, std::vector<double> const &joint_positions,
+    std::vector<std::string> const &link_names) const {
+  auto state = std::make_shared<robot_state::RobotState>(model_);
+  return getLinkToRobotTransforms(joint_names, joint_positions, state, link_names);
+}
+
+std::vector<std::vector<Eigen::Matrix4Xd>> JacobianFollower::batchGetLinkToRobotTransforms(
+    std::vector<std::vector<std::string>> const &joint_names, std::vector<std::vector<double>> const &joint_positions,
+    std::vector<std::string> const &link_names) const {
+  auto state = std::make_shared<robot_state::RobotState>(model_);
+  std::vector<std::vector<Eigen::Matrix4Xd>> transforms;
+  auto const batch_size = joint_names.size();
+  for (auto b{0u}; b < batch_size; ++b) {
+    auto const joint_names_b = joint_names[b];
```

Comment:
1) Remove these intermediate allocations. `transforms.emplace_back(getLinkToRobotTransforms(joint_names[b], joint_positions[b], state, link_names));` reads just as cleanly, and doesn't require your two lines `auto const joint_names_b = joint_names[b];`
2) If you really want these intermediate variables for some reason, they can be references `auto const &joint_names_b = joint_names[b];`

Reply:
I'd be $100 the compiler removes the allocation & copy
but still I agree with the suggested change